overlord/devicestate/handlers.go

```diff
@@ -76,7 +77,21 @@ func isSameAssertsRevision(err error) bool {
 	return false
 }
 
-func (m *DeviceManager) doSetModel(t *state.Task, _ *tomb.Tomb) error {
+type SetModelInjected func(t *state.Task) error
+
+var (
+	setModelInjected   SetModelInjected
+	setModelInjectedMu sync.Mutex
+)
+
+// InjectedSetModel is only useful for testing
+func InjectSetModel(f SetModelInjected) {
+	setModelInjectedMu.Lock()
```

Comment:
why the lock given that we don't use it when reading?

I would just go with:

var injectedSetModelError error

and

func InjectSetModelError(err error) {
}